packages/electron-updater/src/providers/GitHubProvider.ts

```diff
@@ -77,7 +106,7 @@ export class GitHubProvider extends BaseGitHubProvider<GithubUpdateInfo> {
       throw newError(`No published versions on GitHub`, "ERR_UPDATER_NO_PUBLISHED_VERSIONS")
     }
 
-    const channelFile = getChannelFilename(this.getDefaultChannelName())
+    const channelFile = getChannelFilename(this.updater.allowPrerelease ? this.getCustomChannelName(String(semver.prerelease(tag)?.[0] || 'latest')) : this.getDefaultChannelName())
```

Comment:
So, this change ends up breaking backward compatibility with previous prereleases that still use the `latest.yml` tag.
The unit test fails for a current repo: https://github.com/develar/__test_nsis_release/releases/tag/v1.5.2-beta.3
```
HttpError: 404 Not Found
    "method: GET url: https://github.com/develar/__test_nsis_release/releases/download/v1.5.2-beta.3/beta.yml
```
I'm trying to brainstorm ways we can properly handle this. One way is to have `latest` always be a fallback to any release, so `allowPrerelease` would allow updates out of the alpha/beta cells to prod. What are your thoughts?

Reply:
I think something like this could work?
```
const fetchData = (channelName: string) => {
    const channelFile = getChannelFilename(channelName)
    const channelFileUrl = newUrlFromBase(this.getBaseDownloadPath(tag, channelFile), this.baseUrl)
    const requestOptions = this.createRequestOptions(channelFileUrl)
    try {
        return (await this.executor.request(requestOptions, cancellationToken))!
    } catch (e) {
        if (e instanceof HttpError && e.statusCode === 404) {
            throw newError(`Cannot find ${channelFile} in the latest release artifacts (${channelFileUrl}): ${e.stack || e.message}`, "ERR_UPDATER_CHANNEL_FILE_NOT_FOUND")
        }
        throw e
    }
}

try {
    const channel = this.updater.allowPrerelease ? this.getCustomChannelName(String(semver.prerelease(tag)?.[0] || 'latest')) : this.getDefaultChannelName()
    rawData = fetchData(channel)
} catch (e) {
    if (this.updater.allowPrerelease) {
        // Allow fallback to `latest.yml`
        rawData = fetchData(this.getDefaultChannelName())
    } else {
        throw e
    }
}
```